validator_client/src/sync_committee_service.rs

```diff
@@ -214,28 +214,30 @@ impl<T: SlotClock + 'static, E: EthSpec> SyncCommitteeService<T, E> {
     ) -> Result<(), ()> {
         let log = self.context.log().clone();
 
-        let committee_signatures = validator_duties
-            .iter()
-            .filter_map(|duty| {
-                self.validator_store
-                    .produce_sync_committee_signature(
-                        slot,
-                        beacon_block_root,
-                        duty.validator_index,
-                        &duty.pubkey,
-                    )
-                    .map_err(|e| {
-                        crit!(
-                            log,
-                            "Failed to sign sync committee signature";
-                            "validator_index" => duty.validator_index,
-                            "slot" => slot,
-                            "error" => ?e,
-                        );
-                    })
-                    .ok()
-            })
-            .collect::<Vec<_>>();
+        let mut committee_signatures = Vec::with_capacity(validator_duties.len());
+        for duty in &validator_duties {
+            match self
+                .validator_store
+                .produce_sync_committee_signature(
+                    slot,
+                    beacon_block_root,
+                    duty.validator_index,
+                    &duty.pubkey,
+                )
+                .await
```

Comment:
It just occurred to me that we probably shouldn't be `await`ing these futures in a loop like this.

Unlike the attestation service the sync committee service uses one task for publishing all the signatures. If the VC is managing multiple sync committee members then it will block each sync signature request to the remote signer on the previous one. We could do it in parallel by collecting the futures and joining them.